### PR TITLE
Fix e2e tests when using CRIO 1.28

### PR DESCRIPTION
**Pull request by acardace**
**What this PR does / why we need it**:
1. Fixes this flake https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/10884/pull-kubevirt-e2e-k8s-1.28-sig-compute/1737261845598703616 by making sure kernelboot containers are ready before trying to mount kernel artifacts into virt-launcher.
2. Removes a useless e2e test which will always fail when run with CRIO 1.28.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/container-disk/mount_test.go [hunk 1]
```diff
@@ -81,37 +81,95 @@ var _ = Describe("ContainerDisk", func() {
...
+				HaveOccurred(),
+				false,
+			),
+			Entry("return true and no error once everything is ready and we are within the tolerated retry period",
+				func(*v1.VirtualMachineInstance, int) (string, error) { return "someting", nil },
+				time.Duration(0),
+				Succeed(),
+				true,
+			),
+			Entry("should return true and no error once everything is ready when we are outside of the tolerated retry period",
```
Comment (jean-edouard) [inline pkg/virt-handler/container-disk/mount_test.go:114]: Nit: extra `should` :)

File: pkg/virt-handler/container-disk/mount_test.go [hunk 2]
```diff
@@ -81,37 +81,95 @@ var _ = Describe("ContainerDisk", func() {
...
+					HaveOccurred(),
+					false,
+				),
+				Entry("return true and no error once everything is ready and we are within the tolerated retry period",
+					func(*v1.VirtualMachineInstance) (string, error) { return "someting", nil },
+					time.Duration(0),
+					Succeed(),
+					true,
+				),
+				Entry("should return true and no error once everything is ready when we are outside of the tolerated retry period",
```
Comment (jean-edouard) [inline pkg/virt-handler/container-disk/mount_test.go:166]: Extra `should`

**Review by jean-edouard — COMMENTED** [review]
Tiny nit in-line.
Could you please give a quick explanation on why the e2e test is useless and/or why it fails with CRIO 1.28? Thanks!

**Comment by jean-edouard** [comment]
> Tiny nit in-line. Could you please give a quick explanation on why the e2e test is useless and/or why it fails with CRIO 1.28? Thanks!

Nevermind, the commit message told me what I was looking for :)
/approve

**Review by xpivarc — COMMENTED** [review]
/approve

File: pkg/virt-handler/container-disk/mount_test.go (hunk 1 above)
Comment (acardace) [reply pkg/virt-handler/container-disk/mount_test.go:114]: ouch, fixing this.

**Pull request merged**